artemis/modules/classifier.py

```diff
@@ -30,33 +29,27 @@ class Classifier(ArtemisBase):
     def _sanitize(data: str) -> str:
         data = data.lower()
 
-        if "hxxp://" in data or "hxxps://" in data or "[.]" in data:
-            raise RuntimeError(
-                "Defanged URL detected. If you really want to scan it, please provide it as a standard one."
-            )
-
-        # strip URL schemes
-        if "://" in data:
-            hostname = urllib.parse.urlparse(data).hostname
-            assert hostname is not None
-            data = hostname
+        # Strip leading and trailing dots (e.g. domain.com.)
+        data = data.strip(".")
```

Comment:
isn't it URL that is being passed there? If yes, you should retrieve hostname eg using urlparse

Reply:
nope

previously Artemis silently converter URLs and host:port strings to domains

now Artemis explicitely says that these aren't allowed and you need to provide ip/domain